lib/internal/url.js

```diff
@@ -874,32 +871,32 @@ class URLSearchParams {
     this[context] = null;
   }
 
-  [util.inspect.custom](recurseTimes, ctx) {
+  [util.inspect.custom](depth, opts) {
     if (!this || !this[searchParams] || this[searchParams][searchParams]) {
       throw new TypeError('Value of `this` is not a URLSearchParams');
     }
 
-    if (typeof recurseTimes === 'number' && recurseTimes < 0)
-      return ctx.stylize('[Object]', 'special');
+    if (typeof depth === 'number' && depth < 0)
+      return opts.stylize('[Object]', 'special');
 
-    const separator = ', ';
-    const innerOpts = Object.assign({}, ctx);
-    if (recurseTimes !== null) {
-      innerOpts.depth = recurseTimes - 1;
+    var separator = ', ';
```

Comment:
I don't think it's a 'bug' per se, just something that isn't optimized yet in the version of V8 in master.